### PR TITLE
Add Github Action pipelines for automating releases

### PR DESCRIPTION
**Pull request by michaelosthege**
The `preview` pipeline runs after merging to `master`.
These changes then go "live" under `https://rtlive.de/preview` which allows for checking against bugs before releasing through the creation of a Github Release.

The purpose of these additions is to make updating the website more transparent and less manual work.

In addition to the workflow YAMLs, I had to set `alt` texts on the images and set the modal `img` `src` to something to make the HTML checker happy.

### CODE REVIEW

File: .github/workflows/preview.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    steps:
+    - uses: actions/checkout@v2
+    - name: Check HTML
+      uses: chabad360/htmlproofer@master
+      with:
+        directory: "."
+    - name: Deploy preview site
+    - uses: jakejarvis/s3-sync-action@master
+      with:
+        args: --exclude '*' --include '*.html' --include 'assets/*' --include 'global'
```
Comment (michaelosthege) [inline .github/workflows/preview.yml:17]: These are documented here: https://docs.aws.amazon.com/cli/latest/reference/s3/index.html#use-of-exclude-and-include-filters

File: .github/workflows/preview.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  deploy-preview:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Check HTML
+      uses: chabad360/htmlproofer@master
+      with:
+        directory: "."
+    - name: Deploy preview site
+    - uses: jakejarvis/s3-sync-action@master
```
Comment (michaelosthege) [inline .github/workflows/preview.yml:15]: The action README is here: https://github.com/marketplace/actions/s3-sync

File: assets/global.js [hunk 3]
```diff
@@ -59,7 +59,7 @@ function showRegion(country_alpha2, countryName, region, regionName, date) {
     else {
         $('#regionModalTitle').html(`${unescape(countryName)} / ${unescape(regionName)}`);
     }
-    $('#regionModalImage').attr('src', `data/${country_alpha2}/${date}/${region}/details.png`)
+    $('#regionModalImage').attr('src', `${window.location.origin}/data/${country_alpha2}/${date}/${region}/details.png`)
```
Comment (michaelosthege) [inline assets/global.js:62]: `window.location.origin` resolves to `https://rtlive.de` even if the HTML is located in `https://rtlive.de/preview/index.html`.

At the same time it resolves to `http://localhost:4040` when the site is run locally for testing using https://github.com/michaelosthege/rtlive-dash-de/blob/master/nginx/start.sh

Comment (lhelleckes) [reply assets/global.js:62]: What do you mean by "resolves to"? That it takes the files from the rt.live bucket?

Comment (michaelosthege) [reply assets/global.js:62]: This line of JavaScript runs client-side in the browser: `window.location` is just the URL of the current browser tab.

Could be that I mixed up "resolves" and "evaluates"..

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    - uses: jakejarvis/s3-sync-action@master
+      with:
+        args: --exclude '*' --include '*.html' --include 'assets/*' --include 'global'
+      env:
+        AWS_S3_BUCKET: rtlive.de
+        AWS_ACCESS_KEY_ID: ${{ secrets.AWS_KEY_ID }}
+        AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+        AWS_REGION: 'eu-central-1'
+        SOURCE_DIR: '.'
+        DEST_DIR: ''
```
Comment (lhelleckes) [inline .github/workflows/release.yml:28]: This I don't get. What is this empty DEST_DIR?

File: .github/workflows/release.yml [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+jobs:
+  deploy-live:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Check HTML
+      uses: chabad360/htmlproofer@master
+      with:
+        directory: "."
+    - name: Deploy preview site
```
Comment (lhelleckes) [inline .github/workflows/release.yml:18]: This is not for the preview, is it?

File: .github/workflows/release.yml (hunk 4 above)
Comment (michaelosthege) [reply .github/workflows/release.yml:28]: The empty DEST_DIR is the root level of the S3 bucket where the website lives

File: .github/workflows/release.yml (hunk 5 above)
Comment (lhelleckes) [inline .github/workflows/release.yml:18]:
```suggestion
    - name: Deploy site
```

File: .github/workflows/release.yml (hunk 4 above)
Comment (lhelleckes) [reply .github/workflows/release.yml:28]: If I see it right in the documentation, you can skip the argument in that case.

Comment (michaelosthege) [reply .github/workflows/release.yml:28]: True. I just wanted to make it clear that it's intentional to upload to the root of the bucket.

**Review by lhelleckes — APPROVED** [review]
All fine to me! Great improvement, I am sure it will make things much easier in the future :)

**Comment by michaelosthege** [comment]
Thank you for the review, @lhelleckes !

**Pull request merged**